src/dir_contents.mli

```diff
@@ -34,11 +34,19 @@ val lookup_module : t -> Module.Name.t -> Dune_file.Buildable.t option
 (** All mld files attached to this documentation stanza *)
 val mlds : t -> Dune_file.Documentation.t -> Path.t list
 
-val get : Super_context.t -> dir:Path.t -> t
+type get_result =
+  | Standalone_or_root of t
+  | Group_part of Path.t
+
+(** Produces rules for all group parts when it returns [Standalone_or_root].
+    Does not generate any rules when it returns [Group_part]. *)
+val get : Super_context.t -> dir:Path.t -> get_result
+
+val get_without_rules : Super_context.t -> dir:Path.t -> t
 
 type kind = private
   | Standalone
-  | Group_root of t list Lazy.t (** Sub-directories part of the group *)
+  | Group_root of (unit -> t list) (** Sub-directories part of the group *)
```

Comment:
One thing that I dislike about `Memo.lazy_` is that now we don't know if a function is memoized or not. I know  memoization should be transparent, but for anonymous functions it's something you always have to think about it. Would it make sense to have a `'a Memo.Lazy.t` type perhaps?

(This is a bit of a meta comment not to be addressed in this PR)

Reply:
Yeah, I was weakly thinking that too. I'll make that change in a separate PR.